### PR TITLE
Created User entity, related repository, mapper, DTOs, and a registra…

### PR DESCRIPTION
**Pull request by f3ops3**
…tion endpoint.

### CODE REVIEW

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
add ci.yml file first to the project

File: src/main/java/springweb/courseproject/service/user/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto userRegistrationRequestDto)
+            throws RegistrationException {
+        User user = userMapper.toUser(userRegistrationRequestDto);
+        if (userRepository.existsByEmail(user.getEmail())) {
+            throw new RegistrationException("User already exists");
```
Comment (ahoienko) [inline src/main/java/springweb/courseproject/service/user/UserServiceImpl.java:23]:
```suggestion
            throw new RegistrationException("User with email: " + email + " already exists");

```

File: src/main/java/springweb/courseproject/validation/user/EmailValidator.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email != null && Pattern.compile(EMAIL_PATTERN).matcher(email).matches();
+    }
```
Comment (ahoienko) [inline src/main/java/springweb/courseproject/validation/user/EmailValidator.java:13]: Is it necessary to write custom annotation for such logic? Please investigate I`m 99% sure that such annotation already exists

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: last_name
+                  type: VARCHAR(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: VARCHAR(255)
```
Comment (ahoienko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:39]: Add empty line

**Review by ahoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments)

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:40]:
```suggestion
```
only one empty line ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Good job!

**Pull request merged**